readium/shared/src/main/java/org/readium/r2/shared/util/downloads/DownloadManager.kt

```diff
@@ -25,10 +26,16 @@ import org.readium.r2.shared.util.mediatype.MediaType
  */
 public interface DownloadManager {
 
-    public data class Request(
-        val url: AbsoluteUrl,
-        val headers: Map<String, List<String>> = emptyMap()
-    )
+    public class Request private constructor(
```

Comment:
Why prevent giving a custom ID?

Reply:
I am pretty glad with this solution actually. It nicely prevents ID conflicts between the toolkit and the app. What would be the point of custom IDs?